examples/simple-sampler/foo_library/foo_library.h

```diff
@@ -0,0 +1,3 @@
+#pragma once
+
+void foo_library();
```

Comment:
Should this library be within a namespace? Also, method names in C++ OpenTelemetry are CapitalCase (so FooLibrary). Lastly, FooLibrary doesn't quite describe what this function does, so consider renaming to something more descriptive, and also add a comment for what this method does.